Assets/HoloToolkit/SpatialSound/Scripts/AudioInfluencerManager.cs

```diff
@@ -0,0 +1,242 @@
+        [Tooltip("Maximum number of objects that will be considered when looking for influencers.")]
+        [Range(1, 25)]
+        [SerializeField]
+        private int MaxObjects = 10;
+
+        // Time of last audio processing update.
+        private DateTime lastUpdate = DateTime.MinValue;
+
+        // The source of the audio.
+        private AudioSource audioSource;
+
+        // The initial volume level of the audio source.
+        private float initialAudioSourceVolume;
+
+        // The hits returned by Physics.RaycastAll
+        private RaycastHit[] hits;
+
+        // The collection of previously applied audio influencers.
+        private List<IAudioInfluencer> previousInfluencers = new List<IAudioInfluencer>();
+
+        // Potential effects manipulated by an audio influencer and their key
+        // properties.
+        private AudioLowPassFilter lowPassFilter;
+        private float nativeLowPassCutoffFrequency;
+        private AudioHighPassFilter highPassFilter;
+        private float nativeHighPassCutoffFrequency;
+
+        private void Awake() 
+        {           
+            audioSource = gameObject.GetComponent<AudioSource>();
+            initialAudioSourceVolume = audioSource.volume;
+
+            // Get optional filters (and initial values) that the sound designer / developer 
+            // may have applied to this game object
+            lowPassFilter = gameObject.GetComponent<AudioLowPassFilter>();
+            nativeLowPassCutoffFrequency = (lowPassFilter != null) ? lowPassFilter.cutoffFrequency : NeutralHighFrequency;
+            highPassFilter = gameObject.GetComponent<AudioHighPassFilter>();
+            nativeHighPassCutoffFrequency = (highPassFilter != null) ? highPassFilter.cutoffFrequency : NeutralLowFrequency;
+
+            // Preallocate the array that will be used to collect RaycastHit structures.
+            hits = new RaycastHit[MaxObjects];
+        }
+	
+	    private void Update() 
+        {
+            DateTime now = DateTime.Now;
+
+            // Audio influences are not updated every frame.
+            if ((UpdateInterval * 1000.0f) <= (now - lastUpdate).Milliseconds)
+            {
+                audioSource.volume = initialAudioSourceVolume;
+
+                // Get the audio influencers that should apply to the audio source.
+                List<IAudioInfluencer> influencers = GetInfluencers();
+                foreach (IAudioInfluencer influencer in influencers)
+                {
+                    // Apply the influencer's effect.
+                    influencer.ApplyEffect(gameObject);
+                }
+
+                // Find and remove the audio influencers that are to be removed from the audio source.
+                List<IAudioInfluencer> influencersToRemove = new List<IAudioInfluencer>();
+                foreach (IAudioInfluencer prev in previousInfluencers)
+                {
+                    MonoBehaviour mbPrev = prev as MonoBehaviour;
```

Comment:
This seems odd. Let's replace `MonoBehaviour` with `GameObject` instead.

Reply:
actually, I have a better way of doing this that I am coding up now.

Reply:
sure. I can do that.

Reply:
I'll take a look

Reply:
The compiler complains about casting to GameObject via the as keyword (used to avoid InvalidCastException). I will see if there is a better solution than casting to a MonoBehaviour.